sdk/rust/oak/src/lib.rs

```diff
@@ -37,7 +37,16 @@ use serial_test_derive::serial;
 /// Handle used to identify read or write channel halves.
 ///
 /// These handles are used for all host function calls.
-pub type Handle = u64;
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct Handle {
```

Comment:
FTR, an alternative here may be a "newtype", e.g. `pub struct Handle(u64)`, but it's probably worth keeping the proper struct (as you have done), in case we want to add more metadata, e.g. type of handle or "rights" to the underlying object.

See https://doc.rust-lang.org/rust-by-example/generics/new_types.html

Reply:
Yeah, since the id is private, changing this would not require a large change either way.